pytket/extensions/qiskit/backends/ibm.py

```diff
@@ -370,7 +396,12 @@ def default_compilation_pass(
         # https://cqcl.github.io/pytket-qiskit/api/index.html#default-compilation
         # Edit this docs source file -> pytket-qiskit/docs/intro.txt
         if optimisation_level == 0:
-            if self._standard_gateset:
+            if self._primitive_gates == {OpType.X, OpType.SX, OpType.Rz, OpType.CX} or {
```

Comment:
This rather messy if statement is to avoid a test failure [here](https://github.com/CQCL/pytket-qiskit/blob/1110d9e096b4bf0ae4788d844763ee89fb1bb9d1/tests/backend_test.py#L1109)

I was wondering why the `if self._standard_gateset` check was there before.

Basically its there to ensure that the rebase is skipped in the default compilation pass if its not {X, SX, Rz, CX}.

The `test_cloud_stabiliser` test compiles a circuit that already satisfies the `GateSetPredicate` for the sim so the rebase should be skipped altogether as we can't find a valid TK1 decomposition.

This confused me for quite a while